### PR TITLE
Topic/parallelize selftests pr

### PR DESCRIPTION
**Pull request by danielocfb**
This pull request proposes the necessary changes for running the selftests in individual jobs and possibly in parallel to each other. Please look at individual commits, all of which can stand on their own.

### CODE REVIEW

**Comment by danielocfb** [comment]
This pull request requires https://github.com/libbpf/ci/pull/32 first.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -22,23 +22,35 @@ jobs:
...
+
+          matrix = [
+            {"kernel": "LATEST", "runs_on": ["ubuntu-latest", "self-hosted"], "arch": "x86_64", "toolchain": "gcc"},
+            {"kernel": "LATEST", "runs_on": ["ubuntu-latest", "self-hosted"], "arch": "x86_64", "toolchain": "${{ needs.llvm-toolchain.outputs.llvm }}"},
+            {"kernel": "LATEST", "runs_on": ["z15", "self-hosted"], "arch": "s390x", "toolchain": "gcc"},
+          ]
+          build_matrix = {"include": matrix}
+          print(f"::set-output name=build_matrix::{dumps(build_matrix)}")
+
+          tests = ["test_progs", "test_progs_noalu", "test_maps", "test_verifier"]
```
Comment (anakryiko) [inline .github/workflows/test.yml:41]: as discussed, probably good to s/test_progs_noalu/test_progs_noalu32/ while it's not too late to keep it close to actual binary name

**Review by anakryiko — APPROVED** [review]
Exciting!

File: .github/workflows/test.yml (hunk 1 above)
Comment (danielocfb) [reply .github/workflows/test.yml:41]: I've renamed the test here https://github.com/kernel-patches/vmtest/pull/129.

**Comment by danielocfb** [comment]
I renamed the jobs to be a bit more expressive, as discussed in person.

**Comment by danielocfb** [comment]
Remaining CI failures are unrelated (patch has been submitted upstream).

**Pull request merged**